File: CapstoneProjectAPI/Models/DTOs/ApprovalActionResponseDto.cs
using System;

namespace CapstoneProjectAPI.Models.DTOs
{
    public class ApprovalActionResponseDto
    {
        public int Id { get; set; }
        public int ApproverUserId { get; set; }
        public string ApproverUserName { get; set; } = string.Empty;
        public string Action { get; set; } = string.Empty;
        public string? Comments { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
    }
}

File: CapstoneProjectFrontend/src/app/dtos/approval-action.response.dto.ts
import { ApprovalActionType } from "../enums/approval-action-type.enum";

export interface ApprovalActionResponseDto {
  id: number;
  approverUserId: number;
  approverUserName: string;
  action: ApprovalActionType | string; 
  comments: string | null;             
  createdAt: string;                   
}